internal/datacoord/services.go

```diff
@@ -515,6 +515,12 @@ func (s *Server) DropVirtualChannel(ctx context.Context, req *datapb.DropVirtual
 	}
 
 	log.Info("DropVChannel plan to remove", zap.String("channel", channel))
+	err = s.meta.DropChannelCheckpoint(channel)
+	if err != nil {
+		log.Error("DropChannelCheckpoint failed", zap.String("vChannel", channel), zap.Error(err))
+		resp.Status.Reason = err.Error()
+		return resp, nil
+	}
```

Comment:
Please don't change meta here, if DC fails after the channel meta change, it'll never recover properly.

Please move these lines into `meta.FinishDropChannel` to make it right.

Reply:
you're right